ledger/heavy/executor/jetkeeper.go

```diff
@@ -284,31 +349,78 @@ func infoToList(s map[insolar.JetID]struct{}) []insolar.JetID {
 	return r
 }
 
-func (jk *dbJetKeeper) checkPulseConsistency(ctx context.Context, pulse insolar.PulseNumber) bool {
-	expectedJets := jk.jetTrees.All(ctx, pulse)
+func (jk *dbJetKeeper) getTopSyncJets() ([]insolar.JetID, error) {
+	var result []insolar.JetID
+	top := jk.topSyncPulse()
+	if top == insolar.FirstPulseNumber {
+		return []insolar.JetID{insolar.ZeroJetID}, nil
+	}
+	jets, err := jk.get(top)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't getTopSyncJets: %d", top)
+	}
+
+	for _, ji := range jets {
+		if *ji.Split {
+			left, right := jet.Siblings(ji.JetID)
+			result = append(result, left, right)
+		} else {
+			result = append(result, ji.JetID)
+		}
+	}
+
+	return result, nil
+
+}
+
+func compareJets(what []insolar.JetID, actualJetsSet map[insolar.JetID]struct{}) error {
+	if len(actualJetsSet) != len(what) {
+		if len(actualJetsSet) > len(what) {
+			return errors.New("num actual jets is more then topSyncJets." +
+				" TopSyncJets: " + insolar.JetIDCollection(what).DebugString() +
+				". Actual: " + insolar.JetIDCollection(infoToList(actualJetsSet)).DebugString())
+		}
+	}
+
+	for _, expID := range what {
+		if _, ok := actualJetsSet[expID]; !ok {
+			return errors.New("jet sets are different. it's too bad. " +
+				". TopSyncJets: " + insolar.JetIDCollection(what).DebugString() +
+				". Actual: " + insolar.JetIDCollection(infoToList(actualJetsSet)).DebugString())
+		}
+	}
+
+	return nil
+}
+
+func (jk *dbJetKeeper) checkPulseConsistency(ctx context.Context, pulse insolar.PulseNumber, checkBackup bool) bool {
+	logger := inslogger.FromContext(ctx)
+	topSyncJets, err := jk.getTopSyncJets()
+	if err != nil {
+		logger.Error("can't get jets for top sync pulse: ", err)
+		return false
+	}
 	actualJets := jk.all(pulse)
 
```

Comment:
Should always return false if topSyncPulse != pulse.Prev.

Reply:
done